src/interval.jl

```diff
@@ -288,18 +295,26 @@ function Base.union!(intervals::Union{AbstractVector{<:Interval}, AbstractVector
     return intervals
 end
 
+"""
+    superset(intervals::AbstractArray{<:AbstractInterval}) -> Interval
+
+Create a single interval which encompasses all of the provided intervals.
+"""
+function superset(intervals::AbstractArray{<:AbstractInterval})
+    left = minimum(Intervals.LeftEndpoint.(intervals))
```

Comment:
It should be more efficient to only make one pass over the array by providing the function as an argument to `maximum` and `minimum`, e.g.
```julia
left = minimum(LeftEndpoint, intervals)
right = maximum(RightEndpoint, intervals)
```
(Also note that the module prefix shouldn't be necessary within the module.)

Reply:
Thanks. I was developing this in the REPL which is why I have the module prefix

Reply:
Thanks. I was developing this in the REPL which is why I have the module prefix

Reply:
That's... really weird. 🤔